fpdf/graphics_state.py

```diff
@@ -57,11 +57,14 @@ def __init__(self, *args, **kwargs):
         ]
         super().__init__(*args, **kwargs)
 
-    def _push_local_stack(self):
-        self.__statestack.append(self.__statestack[-1].copy())
+    def _push_local_stack(self, new=None):
```

Comment:
This mean that _stack entries_ are now a concept that can be used **outside** that class?

Then maybe it would be worth to define a `@dataclass` type for them, not just use a `dict()`.

What do you think?

Reply:
They aren't really meant to be _manipulated_ elsewhere. I just needed a way to temporarily step down one level (to send stuff to the output), and then return back to where I was for processing more input. Re-adding the popped item was the simplest way I could think of to do that.
I don't think that a data class would make any difference for that, so it's probably not worth the overhead.

Reply:
I think you are using this only in `HTML2FPDF._end_paragraph()`, and it's really clear to me why those pop/push operations are needed around the call to `self._column.render()`...

Reply:
It is necessary because `_render_styled_text_line()` decides to wrap the line inside a q/Q pair depending on whether the rendered text changed any of the settings relative to the current stack level. Write_html() uses another stack level to isolate its local changes. And since we intermittingly process input and write output, that makes it necessary to jump between the two.